crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -155,6 +155,76 @@ impl<P: FsProvider> ResctrlPlugin<P> {
             warn!("resctrl-plugin: failed to send event: {}", e);
         }
     }
+
+    fn should_emit_and_update(
+        &self,
+        pod_uid: &str,
+        new_state: AssignmentState,
+        group_path: Option<String>,
+    ) -> Option<PodResctrlEvent> {
+        let mut st = self.state.lock().unwrap();
+        let ps = st.pods.entry(pod_uid.to_string()).or_default();
+
+        let changed = ps.last_state.as_ref() != Some(&new_state) || ps.group_path != group_path;
+        if changed {
+            ps.last_state = Some(new_state.clone());
+            ps.group_path = group_path.clone();
+            Some(PodResctrlEvent::Added(PodResctrlAdded {
+                pod_uid: pod_uid.to_string(),
+                group_path,
+                state: new_state,
+            }))
+        } else {
+            None
+        }
+    }
+
+    fn enumerate_container_pids(&self, container: &nri::api::Container) -> Vec<i32> {
```

Comment:
let's create a CgroupPidSource trait and implement one for real systems and a mock one for testing (like we do for FS in the resctrl crate. For the real system, we should use https://crates.io/crates/cgroups-rs